src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -1894,5 +1882,467 @@ struct S
+                    );
+        }
+
+        [Fact]
+        public void Subsumption_02()
+        {
+            var src = @"
+class C
+{
+    void Test(int[] a, int[] b)
+    {
+        switch (a, b)
+        {
+            case ([.., 42], [.., 43]):
+            case ([42], [43]):
+                break;
+        }
+    }
+}";
+            var comp = CreateCompilationWithIndexAndRange(src, parseOptions: TestOptions.RegularWithListPatterns);
+            comp.VerifyEmitDiagnostics(
+                // (9,18): error CS8120: The switch case is unreachable. It has already been handled by a previous case or it is impossible to match.
+                //             case ([42], [43]):
+                Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "([42], [43])").WithLocation(9, 18));
+        }
+
+        [Fact]
+        public void Subsumption_03()
+        {
+            var src = @"
+class C
+{
+    void Test(int[] a)
+    {
+        switch (a)
+        {
+            case { Length: 1 } and [.., 1]:
+            case { Length: 1 } and [1, ..]:
+                break;
+        }
+        switch (a)
+        {
+            case { Length: 1 } and [1, ..]:
+            case { Length: 1 } and [.., 1]:
+                break;
+        }
+    }
+}";
+            var comp = CreateCompilationWithIndexAndRange(src, parseOptions: TestOptions.RegularWithListPatterns);
+            comp.VerifyEmitDiagnostics(
+                    // (9,18): error CS8120: The switch case is unreachable. It has already been handled by a previous case or it is impossible to match.
+                    //             case { Length: 1 } and [1, ..]:
+                    Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "{ Length: 1 } and [1, ..]").WithLocation(9, 18),
+                    // (15,18): error CS8120: The switch case is unreachable. It has already been handled by a previous case or it is impossible to match.
+                    //             case { Length: 1 } and [.., 1]:
+                    Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "{ Length: 1 } and [.., 1]").WithLocation(15, 18)
+                    );
+        }
+
+        [Fact]
+        public void Subsumption_04()
+        {
+            var src = @"
+class C
+{
+    void Test(int[] a)
+    {
+        switch (a)
+        {
+            case [1, .., 3]:
+            case [1, 2, 3]:
+                break;
+        }
+    }
+}";
+            var comp = CreateCompilationWithIndexAndRange(src, parseOptions: TestOptions.RegularWithListPatterns);
+            comp.VerifyEmitDiagnostics(
+                // (9,18): error CS8120: The switch case is unreachable. It has already been handled by a previous case or it is impossible to match.
+                //             case [1, 2, 3]:
+                Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "[1, 2, 3]").WithLocation(9, 18)
+                );
+        }
+
+        [Fact]
+        public void Subsumption_05()
```

Comment:
For all the new tests that don't have errors, let's also test execution. We can make the test a theory and verify which case was picked for different inputs.
For more interesting tests (nested slices for example), I would go one test further and verify execution order (which properties/methods are invoked).

Reply:
One more test idea:

```
        [Fact]
        public void Subsumption_TODO2()
        {
            var src = @"
class C
{
    void Test(int[] a)
    {
        _ = a switch
        {
            { Length: not 1 }  => 0,
            [<0, ..] => 0,
            [..[>= 0]] => 1,
            [var unreachable] => 2, // missing diagnostic?
        };
    }
}" + TestSources.GetSubArray;
            var comp = CreateCompilationWithIndexAndRange(src);
            comp.VerifyEmitDiagnostics();
        }

```

Reply:
Added a test and marked as prototype. I *think* the case `[..null]` shouldn't affect exhaustiveness.

Reply:
Actually, it doesn't (as your test shows). Subsumption checking would still consider nulls.

Reply:
Added both to Subsumption_12.